src/main/java/com/commercetools/sync/services/CartDiscountService.java

```diff
@@ -15,6 +15,26 @@
 import java.util.concurrent.CompletionStage;
 
 public interface CartDiscountService {
+
+    /**
+     * Given a {@code key}, this method first checks if a cached map of CartDiscount keys -&gt; ids is not empty.
```

Comment:
Implementation is different than what you describe here. Your code:
```
if (keyToIdCache.containsKey(key)) {
```
vs your comment:
`this method first checks if a cached map of CartDiscount keys -> ids is not empty.`

Reply:
you are right,  this is also missing on some other services, I will open an issue for this and link to this comment.
Commit: 1829967

Reply:
https://github.com/commercetools/commercetools-sync-java/issues/397